test/distributed/_composable/fsdp/test_fully_shard_training.py

```diff
@@ -233,6 +239,21 @@ def test_train_parity_multi_group(self):
             self._test_train_parity_multi_group,
         )
 
+    @skip_if_lt_x_gpu(2)
+    @test_compiled_fsdp()
+    def test_train_parity_multi_group_compile(self):
+        self.run_subtests(
+            {
+                "reshard_after_forward": [True, False],
+                "device_type": ["cuda"],
+                "delay_after_forward": [False, True],
+                "delay_before_all_gather": [False],
+                "delay_before_reduce_scatter": [False],
```

Comment:
Are we only using `False` here to save test time?

Reply:
there is a compiling issue for delayed all-gather/reduce-scatter. it's worth investigating (similar to reshard_after_forward=2)
``` File "/home/weif/local/pytorch/torch/distributed/_composable/fsdp/_fsdp_state.py", line 142, in _pre_forward
    args, kwargs = self._fsdp_param_group.pre_forward(module, args, kwargs)
  File "/home/weif/local/pytorch/torch/distributed/_composable/fsdp/_fsdp_param_group.py", line 212, in pre_forward
    self.wait_for_unshard()
  File "/home/weif/local/pytorch/torch/distributed/_composable/fsdp/_fsdp_param_group.py", line 175, in wait_for_unshard
    foreach_all_gather_copy_out(
  File "/home/weif/local/pytorch/torch/distributed/_composable/fsdp/_fsdp_collectives.py", line 80, in foreach_all_gather_copy_out
    all_gather_work.wait()
AttributeError: 'Tensor' object has no attribute 'wait'
```